cmd/probe/tcp_probe_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+)
+
+func TestTCPProbe(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), false)
+
+		if err := p.Run(t.Context()); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("conn should fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionSucceeded) {
+			t.Fatalf("expecting error %v, got %v", errConnectionSucceeded, err)
+		}
+	})
+
+	t.Run("conn should not fail", func(t *testing.T) {
+		l, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			t.Fatalf("unexpected error creating listener: %v", err)
+		}
+
+		p := NewTCPProbe(l.Addr().String(), true)
+
+		l.Close()
+
+		if err := p.Run(t.Context()); !errors.Is(err, errConnectionFailed) {
+			t.Fatalf("expecting error %v, got %v", errConnectionFailed, err)
```

Comment:
This test is not generating the error being asserted, I'm getting this:
```
--- FAIL: TestTCPProbe (0.00s)
    --- FAIL: TestTCPProbe/conn_should_not_fail (0.00s)
        tcp_probe_test.go:48: expecting error connection failed, got <nil>
FAIL
```

... however, based on the `t.Run` description I'm not sure that this test is supposed to fail. What's the intent of this test? :)

Judging by the code, I think `nil` is expected in this scenario.

Reply:
Same as before, forgot to include the fix for this test. The intent was to confirm that if create a probe that *shouldn't* fail, then when the connections fails we return the right error.